src/g3log/logmessage.hpp

```diff
@@ -87,7 +88,8 @@ namespace g3 {
       // Complete access to the raw data in case the helper functions above
       // are not enough.
       //
-      std::time_t _timestamp;
+      // TODO: unique_ptr should be considered instead of shared_ptr
+      std::shared_ptr<Timestamp> _timestamp;
```

Comment:
Further. See my "broken API" comment below.

I will take another look and ponder this. I am not sure how the templated version would work. The sinks should be ignorant over the time type.

At worst this will not be merged and you create your own fork of g3log with a high precision timer. I can then happily point to your g3log version when other users need such high precision.
---